pkg/deploy/template.go

```diff
@@ -296,11 +298,19 @@ func (c Controller) Template(client Client, template, nameSpace string, payload
 		dispatched.Operation = operation
 		configurationDetails := &Configuration{Action: operation, DeploymentName: dispatched.DeploymentName, InstanceID: dispatched.InstanceID, NameSpace: nameSpace}
 		c.ConfigurationController.Configure(client, configurationDetails)
-		return instansiateBuild()
+		return instansiateBuild(dispatched)
 	}
 	if err := c.statusPublisher.Publish(statusKey, configInProgress, "service does not exist creating"); err != nil {
 		c.Logger.Error("failed to publish status key " + statusKey + " continuing " + err.Error())
 	}
+	dispatches, err := deployDependencyServices(c, client, osTemplate, nameSpace, payload)
+	if err != nil {
+		c.statusPublisher.Publish(statusKey, configError, err.Error())
+		return nil, err
+	}
+
+	waitForDependencies(client, nameSpace, dispatches, payload)
```

Comment:
probably need a better way to do this as it will block the request being ended right?